storage/innobase/include/mach0data.ic

```diff
@@ -139,7 +139,9 @@ bytes. The most significant byte is at the lowest address.
 @param[in]      b       pointer to 4 bytes to read
 @return 32 bit integer */
 static inline uint32_t mach_read_from_4(const byte *b) {
+#ifndef XTRABACKUP
```

Comment:
this doesn't seem right. Who is passing nullptr? can't we fix the caller? Also if it is nullptr, the below code is disaster.

Reply:
it's not that someone is passing a nullptr. The problem is that we don't have low level call such as ut_a or ut_ad.
We could potentially change it:
```
#ifdef XTRABACKUP
#ifdef UNIV_DEBUG
assert(b);
#endif
#else
ut_a(b);
#endif
```

Reply:
ah, ut_() is the problem :D never guessed it will be suppressed because of that :D

Reply:
in release build, it can be empty. because ut_ad() is debug only

Reply:
right. the preprocessor macro is XTRABACKUP && UNIV_DEBUG

Reply:
I mean, the "#else " part is not necessary and ut_a() will be a problem too right?

Reply:
correct, we don't include files that ut_a or ut_ad are defined. If we want to include them, there is a bunch of other unnecessary dependencies that will have to be resolved.
I would rather go with the preprocessor macros for this one.

Reply:
Fixed